### PR TITLE
Add `QuantumNaturalGradient` optimizer

### LINKED ISSUE #207: Add `QuantumNaturalGradient` to `optimizers`
Creation of the `QuantumNaturalGradient` optimizer class in https://github.com/qiboteam/qiboml/blob/main/src/qiboml/models/optimizers.py, following [the original paper](https://arxiv.org/abs/1909.02108).

The class should follow the same syntax as https://github.com/qiboteam/qiboml/blob/8738034fa244d931a2f6825fbe31f824c68019ea/src/qiboml/models/optimizers.py#L18

It should also work for any parametrized circuit, and one should make use of `qibo`'s [QFIM](https://github.com/qiboteam/qibo/blob/437bf712fadbb09f55cd93005870412c7b73705d/src/qibo/quantum_info/metrics.py#L1097) implementation to calculate the metric tensor associated with the given parametrized circuit.

### PR DESCRIPTION
**Pull request by DoView1**
## Summary
- Add a `QuantumNaturalGradient` optimizer for parametrized `qibo.Circuit` objects.
- Use `qibo.quantum_info.quantum_fisher_information_matrix` to build the QFIM metric tensor and solve the regularized natural-gradient system.
- Support user-provided Euclidean gradients, separate `gradient_kwargs`, and a central finite-difference fallback.
- Add a JAX backend `jacobian` implementation so qibo QFIM can run through the qiboml JAX backend as well as the existing PyTorch/TensorFlow jacobian paths.
- Add optimizer tests, advanced docs, and API reference coverage.

## Validation
- `python -m pytest tests/test_models_optimizers.py -q -o addopts=''`
- `python -m black --check src/qiboml/models/optimizers.py src/qiboml/backends/jax.py tests/test_models_optimizers.py`
- `python -m py_compile src/qiboml/models/optimizers.py src/qiboml/backends/jax.py tests/test_models_optimizers.py`
- `git diff --check`

Closes #207

### CODE REVIEW

File: doc/source/advanced/optimizers.rst [hunk 1]
```diff
@@ -1,5 +1,47 @@
...
+The Quantum Natural Gradient (QNG) optimizer updates variational-circuit
+parameters using the quantum Fisher information matrix as a local metric tensor.
+In ``qiboml``, it is implemented in
+:class:`qiboml.models.optimizers.QuantumNaturalGradient`.
+
+The optimizer works with any parametrized :class:`qibo.Circuit`, a callable loss
+function, and either a user-provided Euclidean gradient function or a central
+finite-difference gradient. The QFIM itself is calculated with
+``qibo.quantum_info.quantum_fisher_information_matrix`` and therefore requires a
+backend that supports the QFIM Jacobian calculation.
```
Comment (renatomello) [inline doc/source/advanced/optimizers.rst]:
```suggestion
backend that supports the QFIM's Jacobian calculation.
```

File: doc/source/advanced/optimizers.rst [hunk 2]
```diff
@@ -1,5 +1,47 @@
+Using the Quantum Natural Gradient Optimizer
+--------------------------------------------
+
+The Quantum Natural Gradient (QNG) optimizer updates variational-circuit
+parameters using the quantum Fisher information matrix as a local metric tensor.
```
Comment (renatomello) [inline doc/source/advanced/optimizers.rst]:
```suggestion
parameters using the quantum Fisher information matrix (QFIM) as a local metric tensor.
```

File: doc/source/advanced/optimizers.rst [hunk 3]
```diff
@@ -1,5 +1,47 @@
...
+``qibo.quantum_info.quantum_fisher_information_matrix`` and therefore requires a
+backend that supports the QFIM Jacobian calculation.
+
+Example usage
+~~~~~~~~~~~~~
+
+.. code-block:: python
+
+    from qibo import Circuit, gates, get_backend, set_backend
+    from qiboml.models.optimizers import QuantumNaturalGradient
```
Comment (renatomello) [inline doc/source/advanced/optimizers.rst:22]:
```suggestion
    from qibo import Circuit, gates, get_backend, set_backend
    from qibo.hamiltonians import XXZ
    from qiboml.models.optimizers import QuantumNaturalGradient
```

File: doc/source/advanced/optimizers.rst [hunk 4]
```diff
@@ -1,5 +1,47 @@
...
+
+    set_backend("qiboml", platform="pytorch")
+    backend = get_backend()
+
+    circuit = Circuit(1)
+    circuit.add(gates.RY(0, theta=0.2))
```
Comment (renatomello) [inline doc/source/advanced/optimizers.rst]:
```suggestion
    nqubits = 2

    hamiltonian = XXZ(nqubits, backend=backend).matrix

    circuit = Circuit(1)
    circuit.add(gates.RY(qubit, theta=backend.random_sample(1)) for qubit in range(nqubits))
    circuit.add(gates.CNOT(qubit, qubit + 1) for qubit in range(nqubits - 1))
    circuit.add(gates.RY(qubit, theta=backend.random_sample(1)) for qubit in range(nqubits))
```

File: doc/source/advanced/optimizers.rst [hunk 5]
```diff
@@ -1,5 +1,47 @@
...
+
+    def loss_fn(circuit, backend):
+        state = backend.execute_circuit(circuit).state()
+        return backend.real(backend.conj(state[1]) * state[1])
```
Comment (renatomello) [inline doc/source/advanced/optimizers.rst]:
```suggestion
    def loss_fn(circuit, backend):
        state = backend.execute_circuit(circuit).state()
        return backend.real(backend.conj(state) @ hamiltonian @ state)
```

File: doc/source/advanced/optimizers.rst [hunk 6]
```diff
@@ -92,4 +134,4 @@ At the end of the run, the following objects are returned:
 - ``losses``: List of losses per epoch.
 - ``final_params``: Final parameters.
 
-Also, one can access the arttributes ``n_calls_loss`` and ``n_calls_gradient``, which store respectively the number of times that the loss and the gradient were computed during the optimization.
\ No newline at end of file
+Also, one can access the arttributes ``n_calls_loss`` and ``n_calls_gradient``, which store respectively the number of times that the loss and the gradient were computed during the optimization.
```
Comment (renatomello) [inline doc/source/advanced/optimizers.rst]:
```suggestion
Also, one can access the attributes ``n_calls_loss`` and ``n_calls_gradient``, which store respectively the number of times that the loss and the gradient were computed during the optimization.
```

File: doc/source/advanced/optimizers.rst [hunk 7]
```diff
@@ -1,5 +1,54 @@
...
+    from qiboml.models.optimizers import QuantumNaturalGradient
+
+    set_backend("qiboml", platform="pytorch")
+    backend = get_backend()
+
+    nqubits = 2
+
+    hamiltonian = XXZ(nqubits, backend=backend).matrix
+
+    circuit = Circuit(1)
```
Comment (renatomello) [inline doc/source/advanced/optimizers.rst]:
```suggestion
    circuit = Circuit(nqubits)
```
